demos/device_defender_for_aws/defender_demo.c

```diff
@@ -0,0 +1,649 @@
+                        defenderResponse ) );
+        }
+    }
+
+    return status;
+}
+/*-----------------------------------------------------------*/
+
+static void publishCallback( MQTTContext_t * pxMqttContext,
+                              MQTTPacketInfo_t * pxPacketInfo,
+                              MQTTDeserializedInfo_t * pxDeserializedInfo )
+{
+    DefenderStatus_t status;
+    DefenderTopic_t api;
+    bool validationResult;
+    MQTTPublishInfo_t * pPublishInfo = pxDeserializedInfo->pPublishInfo;
+    
+    /* Silence compiler warnings about unused variables. */
+    ( void ) pxMqttContext;
+    ( void ) pxPacketInfo;
+
+    status = Defender_MatchTopic( pPublishInfo->pTopicName,
+                                  pPublishInfo->topicNameLength,
+                                  &( api ),
+                                  NULL,
+                                  NULL );
+
+    if( status == DefenderSuccess )
+    {
+        if( api == DefenderJsonReportAccepted )
+        {
+            /* Check if the response is valid and is for the report we published. */
+            validationResult = validateDefenderResponse( pPublishInfo->pPayload,
+                                                         pPublishInfo->payloadLength );
+
+            if( validationResult == true )
+            {
+                LogInfo( ( "The defender report was accepted by the service. Response: %.*s.",
+                           ( int ) pPublishInfo->payloadLength,
+                           ( const char * ) pPublishInfo->pPayload ) );
+                reportStatus = ReportStatusAccepted;
+            }
+        }
+        else if( api == DefenderJsonReportRejected )
+        {
+            /* Check if the response is valid and is for the report we published. */
+            validationResult = validateDefenderResponse( pPublishInfo->pPayload,
+                                                         pPublishInfo->payloadLength );
+
+            if( validationResult == true )
+            {
+                LogError( ( "The defender report was rejected by the service. Response: %.*s.",
+                            ( int ) pPublishInfo->payloadLength,
+                            ( const char * ) pPublishInfo->pPayload ) );
+                reportStatus = ReportStatusRejected;
+            }
+        }
+        else
+        {
+            LogError( ( "Unexpected defender API : %d.", api ) );
+        }
+    }
+    else
+    {
+        LogError( ( "Unexpected publish message received. Topic: %.*s, Payload: %.*s.",
+                    ( int ) pPublishInfo->topicNameLength,
+                    ( const char * ) pPublishInfo->pTopicName,
+                    ( int ) pPublishInfo->payloadLength,
+                    ( const char * ) ( pPublishInfo->pPayload ) ) );
+    }
+}
+/*-----------------------------------------------------------*/
+
+static bool collectDeviceMetrics( void )
+{
+    int status = EXIT_FAILURE;
```

Comment:
```suggestion
    bool status = false;
```

Reply:
Made a commit to update and align the function commit type with demo helper.
https://github.com/aws/amazon-freertos/pull/2772/commits/e9d96e299adae03bbd26951b6282bd5ff99a4a95